src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -4754,6 +4754,14 @@ private unsafe char[] GetCanonicalPath(char[] dest, ref int pos, UriFormat forma
                         end += (_info.Offset.Query - _info.Offset.Path);
                     }
                 }
+
+                // On Unix, escape '\\' in path of file uris to '%5C' canonical form.
+                if (!IsWindowsSystem && _syntax.NotAny(UriSyntaxFlags.ConvertPathSlashes) && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && !IsImplicitFile)
```

Comment:
> You are adding dead code for certain builds.

It's no more dead code than if it was in a different file.

> The compiler is giving you a warning that you are ignoring.

What warning?

> The code isn't readable by a human

What's not readable about "if (!IsWindowsSystem && ...)"?

> What does the code coverage tool tell you about the coverage of this cs file when running in Windows vs Unix?

The code isn't even in the compiled binary.